Remove old GL stuff

fholger opened the pull request.

fholger committed:
Remove ARB program support

fholger committed:
Make uniform transforms mandatory

fholger committed:
Remove r_useBaseVertex

stgatilov commented on renderer/Material.cpp:
```diff
@@ -1562,7 +1532,6 @@ void idMaterial::ParseStage( idLexer &src, const textureRepeat_t trpDefault ) {
 					newStage.vertexProgram = R_FindARBProgram( GL_VERTEX_PROGRAM_ARB, token.c_str() );
```
I think the code for vertex program should be removed too.

fholger replied:
I just deleted that whole #ifdef block. Wouldn't have done much good anymore, anyway.

stgatilov commented on renderer/Material.h:
```diff
@@ -175,17 +175,14 @@ static const int	MAX_FRAGMENT_IMAGES = 8;
 	//idMegaTexture		*megaTexture;		// handles all the binding and parameter setting 
 
-	bool				GLSL;
 	//stgatilov: trying to use new shader framework
```
Comment worth changing? =)

fholger replied:
Worth removing, anyway :)

stgatilov commented on renderer/VertexCache.cpp:
```diff
@@ -334,13 +325,8 @@ void idVertexCache::EndFrame() {
-		WaitForGeoBufferSet( nextListNum );
-	} else {
-		// this is going to stall, but it probably doesn't matter on such old GPUs...
-		qglFinish();
```
To be honest, it's probably worth leaving `r_useFenceSync` in the code. I remember that many players (especially with old hardware) used it to cover some issues they had. If we remove it, they will probably get stuck with their problem.

fholger replied:
Hm, I wonder. The only actual report I can clearly remember was for an old Intel card that was stuck on GL 3.1 plus extensions, so that card is out going forward, anyway. The remaining reports are less clear, because unfortunately someone usually posts a collection of possible cvars including this particular one, and it then gets rather difficult to track what actually helped.

Given that fence syncs are core since GL 3.2 and such a vital feature, I'd rather like to get rid of the switch and see if there are really any cards left that claim to implement GL 3.3, but have issues with this. Perhaps we could then even figure out what is actually the issue and perhaps fix that or work around it instead of just falling back on the awful glFinish.

fholger committed:
Remove dead code

Commit:
#5349. Implemented SHA256 checksum verification for downloaded missions.
Of course, it is done only if XML file with list of mission contains "sha256" attribute on "downloadLocation" or "localisationPack" element. 

idlib/hashing/sha256.*
  Since SHA256 is computed on-the-fly during download, and download itself is much slower anyway, I took simple standalone but unoptimized implementation of SHA256:
    https://github.com/B-Con/crypto-algorithms
LICENSE.txt:
  Since author wants acknowledgement, added it =)

game/Http/HttpRequest.*
  Added EnableSha256 and GetSha256 methods.
  Basically, if you call the first method, then checksum will be computed.

game/Missions/Download.*
  Added method VerifySha256Checksum, which receives expected sha256 and enables its validation when download is over.
  Added MALFORMED status for the case when checksum verification failed (want special error message).
  Also removed excessive constructor.

game/Missions/MissionManager.*
  LoadModListFromXml now reads and checks "sha256" field.
  Validation is perhaps not necessary, but better let it be.
  For that reason, LoadModListFromXml now returns bool (false if malformed XML).
  Also added MALFORMED status for requests by external users.

game/DownloadMenu.cpp:
  Enabled SHA256 verification for mission/localization packs in CDownloadMenu::StartDownload.
  Print custom error message if mission has wrong checksum.
  Print custom error message if missionlist XML is malformed.
  (the other requests just use error message for FAILED, since they don't return MALFORMED)


git-svn-id: https://svn.thedarkmod.com/svn/darkmod_src/trunk@9022 49c82d7f-2e2a-0410-a16f-ae8f201b507f
